### PR TITLE
Fixing and adding a check for Databricks vs HDInsight when saving and resolving the spark job params

### PR DESCRIPTION
**Pull request by s-tuli**
Fixing and adding a check for Databricks vs HDInsight when saving and resolving the spark job params

### CODE REVIEW

File: Services/DataX.Flow/DataX.Flow.Common/Helper.cs [hunk 1]
```diff
@@ -56,7 +56,7 @@ public static string GetSecretFromKeyvaultIfNeeded(string value)
         /// <returns>true if it is a secret, otherwise false</returns>
         public static bool IsKeyVault(string value)
         {
-            return (value.StartsWith(GetKeyValutNamePrefix()) || value.StartsWith(GetSecretScopePrefix()));
+            return value.StartsWith(GetKeyValutNamePrefix()) || value.StartsWith(GetSecretScopePrefix());
```
Comment (carlbrochu) [inline Services/DataX.Flow/DataX.Flow.Common/Helper.cs]: why change this one?

File: Services/DataX.Config/DataX.Config/ConfigGeneration/Processor/S500_ResolveSparkJobParams.cs [hunk 2]
```diff
@@ -26,10 +27,17 @@ public class ResolveSparkJobParams: ProcessorBase
         public const string TokenName_DatabricksToken = "guiSparkDatabricksToken";
         public const string TokenName_SparkJobDatabricksAutoScale = "guiSparkJobDatabricksAutoScale";
 
+        [ImportingConstructor]
+        public ResolveSparkJobParams(ConfigGenConfiguration conf)
```
Comment (carlbrochu) [inline Services/DataX.Config/DataX.Config/ConfigGeneration/Processor/S500_ResolveSparkJobParams.cs]: does the call to the ctor need to change?

File: Services/DataX.Flow/DataX.Flow.Common/Helper.cs (hunk 1 above)
Comment (s-tuli) [reply Services/DataX.Flow/DataX.Flow.Common/Helper.cs]: I was seeing a formating error, hence fixing it in this checkin

File: Services/DataX.Config/DataX.Config/ConfigGeneration/Processor/S500_ResolveSparkJobParams.cs (hunk 2 above)
Comment (s-tuli) [reply Services/DataX.Config/DataX.Config/ConfigGeneration/Processor/S500_ResolveSparkJobParams.cs]: No caller to the constructor.as such.

**Pull request merged**